compiler/codegen/library.cpp

```diff
@@ -226,7 +227,16 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
   if (requires != "") {
     fprintf(makefile.fptr, "%s", requires.c_str());
   }
-  fprintf(makefile.fptr, " %s\n", libraries.c_str());
+
+  // Erase trailing newline if needed for the workaround below.
+  if (libraries.size() > 0 && *libraries.rbegin() == '\n') {
+    libraries.erase(libraries.end() - 1);
+  }
+
+  fprintf(makefile.fptr, " %s", libraries.c_str());
+
+  // GNU linker won't be able to see config symbols without this.
+  fprintf(makefile.fptr, " %s\n\n", libname.c_str());
```

Comment:
I would like this to only occur for LLVM, if possible.

Also, since you are now modifying something that gets called by both backends, you need to test both backends - which means you need to run test/interop/C/makefiles with CHPL_LLVM=none

Reply:
Are the tests in the `../makefiles` directory sufficient for this? I was under the impression they are equivalent except one runs with the `--llvm` flag.

Reply:
Yes and no.  The directory itself has a skipif for CHPL_LLVM != none, so they won't normally run in that setting (even though they should be equivalent).  And I'd rather be a little cautious than rely on an assumption that is false